### PR TITLE
feat: sentry integration

### PR DESCRIPTION
**Pull request by soren5**
Followed instructions from:
https://github.com/getsentry/sentry-laravel
https://docs.sentry.io/clients/php/integrations/laravel/

to configure sentry and added some instructions about setup in the readme.md

Also implemented sentry reporting for uncaught api exceptions

### CODE REVIEW

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -39,6 +39,10 @@ class Handler extends ExceptionHandler
      */
     public function report(Exception $exception)
     {
+        if (app()->bound('sentry') ) {
```
Comment (hmatalonga) [inline app/Exceptions/Handler.php]: I was wondering why did you not include `&& $this->shouldReport($exception)` to the if statement?

File: app/Exceptions/Handler.php [hunk 2]
```diff
@@ -83,16 +85,13 @@ public function render($request, Exception $exception)
...
-                    ], 500);
                 }
-                // no break
-            default:
-                return response()->json([
-                    'message' => 'Interval server error.',
-                ], 500);
+                break;
         }
+        $exception = new HttpException(500, 'Whoops!');
```
Comment (hmatalonga) [inline app/Exceptions/Handler.php]: What was the reason to create a new exception here?

Maybe we could just do?
```php
return parent::render($request, $exception);
```

**Review by hmatalonga — CHANGES_REQUESTED** [review]
Looks good to me :+1: 

Just a couple of minor details and this will be merged!

File: app/Exceptions/Handler.php (hunk 1 above)
Comment (soren5) [reply app/Exceptions/Handler.php]: When I made some testing I noticed that $this->shouldReport did not return anything for error 401 so I thought we needed to define shouldReport through some middleware.

After further testing I believe that when an exception should NOT be reported shouldReport will return nothing (which evaluates as false) and shouldntReport will return 1 (which evaluates as true)

Screenshot of logs for exception 401:

screen shot 2018-09-10 at 11 32 33

screen shot 2018-09-10 at 11 30 32

In conclusion I'll be using shouldReport in the next push

File: app/Exceptions/Handler.php (hunk 2 above)
Comment (soren5) [reply app/Exceptions/Handler.php]: Yes I believe this is correct, I'll include it in the next push

File: app/Exceptions/Handler.php [hunk 3]
```diff
@@ -57,7 +57,7 @@ public function report(Exception $exception)
     public function render($request, Exception $exception)
     {
         if (!$request->is("api/*")) {
-            return parent::render($request, $exception);
+            return parent::report($request, $exception);
```
Comment (hmatalonga) [inline app/Exceptions/Handler.php]: Was there a reason to invoke report? I didn't understand the change...

This line looked fine, it is the `render` function so the call to the parent method was correct.

File: app/Exceptions/Handler.php [hunk 4]
```diff
@@ -88,8 +88,7 @@ public function render($request, Exception $exception)
                 }
                 break;
         }
-        $exception = new HttpException(500, 'Whoops!');
-        parent::render($request, $exception);
+        parent::report($exception);
```
Comment (hmatalonga) [inline app/Exceptions/Handler.php]: I was referring to delegate the default case to the parent just like so
```php
return parent::render($request, $exception);
```

**Review by hmatalonga — CHANGES_REQUESTED** [review]

File: app/Exceptions/Handler.php (hunk 3 above)
Comment (soren5) [reply app/Exceptions/Handler.php]: I didn't realize that each exception called both render and report so I made the change to make sure all necessary exceptions got reported.
I looked at some other examples of exception handling and I think both methods will get called every time so the change was unnecessary

**Review by hmatalonga — APPROVED** [review]

**Comment by hmatalonga** [comment]
Closes #10
